server/planning/feed_parsers/event_json_tests.py

```diff
@@ -99,3 +102,86 @@ def test_event_json_feed_parser_parse(self):
             assert events[0]["reference"] == "2021/00000001"
             assert events[0]["priority"] == 3
             assert events[0]["language"] == "en"
+
+    def test_event_json_feed_parser_handles_files(self):
+        with self.app.app_context():
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            existing_file_id = ObjectId()
+            self.app.data.insert(
+                "events_files",
+                [
+                    {
+                        "_id": existing_file_id,
+                        "media": "existing-media",
+                        "mimetype": "image/jpeg",
+                    }
+                ],
+            )
+
+            with open(self.sample_json, "r") as f:
+                sample = json.load(f)
+
+            sample["files"] = [str(existing_file_id), "attachments/sunset.jpg"]
+            provider = {"content_expiry": 1}
+
+            with tempfile.NamedTemporaryFile(mode="w", suffix=".json", dir=dir_path, delete=False) as tmp:
+                json.dump(sample, tmp)
+                tmp_path = tmp.name
+
+            events_files_service = get_resource_service("events_files")
+            feeding_service = EventFileFeedingService()
+
+            try:
+                events = EventJsonFeedParser().parse(tmp_path, provider, feeding_service=feeding_service)
+                files = events[0].get("files", [])
+                self.assertEqual(2, len(files))
+                self.assertEqual(existing_file_id, files[0])
+
+                new_file_id = files[1]
+                saved_file = events_files_service.find_one(req=None, _id=new_file_id)
+                self.assertIsNotNone(saved_file)
+                self.assertEqual("image/jpeg", saved_file.get("mimetype"))
+            finally:
+                os.remove(tmp_path)
+
+    def test_event_json_feed_parser_handles_missing_file(self):
+        with self.app.app_context():
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+
+            with open(self.sample_json, "r") as f:
+                sample = json.load(f)
+
+            sample["files"] = ["nonexistent_file.pdf"]
+            provider = {"content_expiry": 1}
+
+            with tempfile.NamedTemporaryFile(mode="w", suffix=".json", dir=dir_path, delete=False) as tmp:
+                json.dump(sample, tmp)
+                tmp_path = tmp.name
+
+            feeding_service = EventFileFeedingService()
+
+            try:
+                events = EventJsonFeedParser().parse(tmp_path, provider, feeding_service=feeding_service)
+                self.assertNotIn("files", events[0])
+            finally:
+                os.remove(tmp_path)
```

Comment:
personally would rather keep it inside that `with tempfile` block and let it delete it, is there any benefit to use this try/finally?